src/test/java/com/hubspot/jinjava/lib/tag/MacroTagTest.java

```diff
@@ -158,7 +158,25 @@ public void itAllowsMacroRecursionWhenEnabledInConfiguration() throws IOExceptio
     }
   }
 
-
+  @Test
+  public void itPreventsRecursionForMacroWithVar() {
```

Comment:
Maybe this test is not necessary, since the recursion cases are tested in the ExpressionNodeTest. This is to confirm that the recursion happens even when macro is called.